KomaMRIBase/src/datatypes/sequence/RF.jl

```diff
@@ -28,19 +56,13 @@ mutable struct RF
     T
     Δf
     delay::Real
-    function RF(A, T, Δf, delay)
-        return if any(T .< 0) || delay < 0
-            error("RF timings must be non-negative.")
-        else
-            new(A, T, Δf, delay)
-        end
-    end
-    function RF(A, T, Δf)
-        return any(T .< 0) ? error("RF timings must be non-negative.") : new(A, T, Δf, 0.0)
-    end
-    function RF(A, T)
-        return any(T .< 0) ? error("RF timings must be non-negative.") : new(A, T, 0.0, 0.0)
-    end
+    center
+    use::RFUse
+    RF(A, T, Δf, delay, center, use) = any(T .< 0) || delay < 0 ? error("RF timings must be non-negative.") : new(A, T, Δf, delay, center, get_RF_use_from_char(use))
+    RF(A, T, Δf, delay, center) = RF(A, T, Δf, delay, center, Undefined())
+    RF(A, T, Δf, delay) = RF(A, T, Δf, delay, 0.0)
+    RF(A, T, Δf) = RF(A, T, Δf, 0.0)
+    RF(A, T) = RF(A, T, 0.0)
```

Comment:
Remove the use of `get_RF_use_from_char` in the constructor.

Don't use recursive definitions. 
```suggestion
    center
    use::RFUse
    RF(A, T, Δf, delay, center, use) = any(T .< 0) || delay < 0 ? error("RF timings must be non-negative.") : new(A, T, Δf, delay, center, use)
    RF(A, T, Δf, delay, center) = RF(A, T, Δf, delay, center, Undefined())
    RF(A, T, Δf, delay) = RF(A, T, Δf, delay, 0.0, Undefined())
    RF(A, T, Δf) = RF(A, T, Δf, 0.0, 0.0, Undefined())
    RF(A, T) = RF(A, T, 0.0, 0.0, 0.0, Undefined())
```